pkg/fleet/installer/installer.go

```diff
@@ -284,37 +293,25 @@ func (i *installerImpl) removePackage(ctx context.Context, pkg string) error {
 	}
 }
 
-func extractPackageLayers(image oci.Image, configDir string, packageDir string) error {
-	layers, err := image.Layers()
-	if err != nil {
-		return fmt.Errorf("could not get image layers: %w", err)
-	}
-	for _, layer := range layers {
-		mediaType, err := layer.MediaType()
+// checkAvailableDiskSpace checks if there is enough disk space at the given paths.
+// This will check the underlying partition of the given path. Note that the path must be an existing dir.
+//
+// On Unix, it is computed using `statfs` and is the number of free blocks available to an unprivileged used * block size
+// See https://man7.org/linux/man-pages/man2/statfs.2.html for more details
+// On Windows, it is computed using `GetDiskFreeSpaceExW` and is the number of bytes available
+// See https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getdiskfreespaceexw for more details
+func checkAvailableDiskSpace(requiredDiskSpace uint64, paths ...string) error {
+	for _, path := range paths {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 Thought:** ‏We never use it for multiple paths... Maybe we should remove that for simplicity. Not a blocker.

Reply:
I'll leave it like that since it's a copy & paste, we can do the change if we have to touch this one